multiversxetl/jobs/transform_job.py

```diff
@@ -0,0 +1,77 @@
+import json
+from pathlib import Path
+from typing import Any, Dict, Protocol
+
+
+class IFileStorage(Protocol):
+    def get_extracted_path(self, task_pretty_name: str) -> Path: ...
+    def get_transformed_path(self, task_pretty_name: str) -> Path: ...
+
+
+class ITask(Protocol):
+    @property
+    def index_name(self) -> str: ...
+    def get_pretty_name(self) -> str: ...
+
+
+class TransformJob:
+    def __init__(self, file_storage: IFileStorage, task: ITask) -> None:
+        self.file_storage = file_storage
+        self.task = task
+        self.transformers: Dict[str, Transformer] = dict()
+
+        self._register_transformer("blocks", BlocksTransformer())
+        self._register_transformer("tokens", TokensTransformer())
+        self._register_transformer("logs", LogsTransformer())
+
+    def _register_transformer(self, index_name: str, transformer: 'Transformer') -> None:
+        self.transformers[index_name] = transformer
+
+    def run(self) -> None:
+        transformer = self.transformers.get(self.task.index_name, Transformer())
+        input_filename = self.file_storage.get_extracted_path(self.task.get_pretty_name())
+        output_filename = self.file_storage.get_transformed_path(self.task.get_pretty_name())
+
+        with open(input_filename) as file:
+            with open(output_filename, "w") as output_file:
+                for line in file:
+                    transformed_line = transformer.transform(line)
+                    output_file.write(transformed_line + "\n")
+
+
+class Transformer:
+    def transform(self, raw_json: str) -> str:
+        data = json.loads(raw_json)
+        data = self._do_transform(data)
+        output = json.dumps(data)
+        return output
+
+    def _do_transform(self, data: Dict[str, Any]) -> Dict[str, Any]:
+        return data
+
+
+class BlocksTransformer(Transformer):
+    def _do_transform(self, data: Dict[str, Any]) -> Dict[str, Any]:
+        del data["pubKeyBitmap"]
+        return data
+
+
+class TokensTransformer(Transformer):
+    def _do_transform(self, data: Dict[str, Any]) -> Dict[str, Any]:
+        data.pop("nft_traitValues", None)
```

Comment:
can we use `data.pop` or `del data[..]` in both places? can remain like it is if not possible

Reply:
Good point. Using `data.pop()` (since it does not throw on missing entries).